### ISSUE TITLE
Bulbs appear in default room with default name after system reboot

### ISSUE THREAD

**Issue by jsve** [ebaauw/homebridge-hue#249]
#### Issue
I have sensors, remotes and bulbs connected via deconz (conbee). Everything works great - until I need to reboot the raspberry for some reason. After a reboot the remotes and sensors keep working, but all bulbs are moved to the default room with the default name. Deconz still identifies the lights as /lights/2 and so on.

In the REST response this is displayed for the first light:
```
	"lights": {
		"2": {
			"ctmax": 454,
			"ctmin": 153,
			"etag": "570ebce0b7e38c506e99bf18460f74d1",
			"hascolor": true,
			"manufacturername": "Philips",
			"modelid": "LTW001",
			"name": "Light 2",
			"state": {
				"alert": "none",
				"bri": 203,
				"colormode": "ct",
				"ct": 250,
				"on": true,
				"reachable": true
			},
			"swversion": "5.50.1.19085",
			"type": "Color temperature light",
			"uniqueid": "00:17:88:01:02:b9:37:d6-0b"
		},
```

config.json:
```
{
      "platform" : "Hue",
      "name" : "Hue",
      "host" : "10.0.1.XX",
      "lights" : true,
      "sensors" : true,
      "users" : {
        "XXX": "YYY"
      }
    },
```

#### Log Messages
```
Jan 02 17:43:36 homebridger homebridge[4364]: [1/2/2018, 5:43:36 PM] [Hue] deCONZ-GW: deCONZ gateway v2.4.99, api v1.0.4
Jan 02 17:43:36 homebridger homebridge[4364]: [1/2/2018, 5:43:36 PM] [Hue] deCONZ-GW: warning: not using recommended deCONZ gateway version 2.4.96
Jan 02 17:43:36 homebridger homebridge[4364]: [1/2/2018, 5:43:36 PM] [Hue] deCONZ-GW: 11 accessories
Jan 02 17:43:36 homebridger homebridge[4364]: [1/2/2018, 5:43:36 PM] [Hue] Initializing platform accessory 'deCONZ-GW'...
Jan 02 17:43:36 homebridger homebridge[4364]: [1/2/2018, 5:43:36 PM] [Hue] deCONZ-GW: 2 services
Jan 02 17:43:36 homebridger homebridge[4364]: [1/2/2018, 5:43:36 PM] [Hue] Initializing platform accessory 'Light 2'...
Jan 02 17:43:36 homebridger homebridge[4364]: [1/2/2018, 5:43:36 PM] [Hue] Initializing platform accessory 'Light 3'...
Jan 02 17:43:36 homebridger homebridge[4364]: [1/2/2018, 5:43:36 PM] [Hue] Initializing platform accessory 'Remote control 1'...
Jan 02 17:43:36 homebridger homebridge[4364]: [1/2/2018, 5:43:36 PM] [Hue] Initializing platform accessory 'Doorbell'...
Jan 02 17:43:36 homebridger homebridge[4364]: [1/2/2018, 5:43:36 PM] [Hue] Initializing platform accessory 'TRADFRI motion sensor '...
Jan 02 17:43:36 homebridger homebridge[4364]: [1/2/2018, 5:43:36 PM] [Hue] Initializing platform accessory 'Temperature 4'...
Jan 02 17:43:36 homebridger homebridge[4364]: [1/2/2018, 5:43:36 PM] [Hue] Initializing platform accessory 'OpenClose 6'...
Jan 02 17:43:36 homebridger homebridge[4364]: [1/2/2018, 5:43:36 PM] [Hue] Initializing platform accessory 'Temperature 7'...
```

**Comment by jsve** [comment]
I'm a bit lost in the ph_ scripts. I did try to replicate its behaviour though. I did a POST to /resourselinks with this body:
```
{
  "name": "homebridge-hue",
  "classid": 1,
  "description": "whitelist",
  "links": [
    "/lights/2",
    "/lights/3"
  ]
}
```

I have only two lights connected at the moment, numbered 2 and 3.

Do I need to do anything else? A reboot I guess, but other than that?

**Also:** does this mean that if the raspberry reboots when one of the lights is disconnected from the grid, homebridge will not start?

**Comment by Krocko** [comment]
@jsve the "description" must be `lightlist`

**Comment by ebaauw** [comment]
> Do I need to do anything else? A reboot I guess, but other than that?

No, that should do the trick.  You might want to run `homebridge -D` just to verify that it works as intended in the logs.  See https://github.com/ebaauw/homebridge-hue#debug-log-file.

Note that I found and fixed another bug so now at v0.5.57.

> the "description" must be `lightlist`

Both `whitelist` and `lightlist` will work.  With `whitelist` the lights will be exposed, irrespective of the config.json settings; with `lightlist` they will only be exposed when specified in config.json.

> Also: does this mean that if the raspberry reboots when one of the lights is disconnected from the grid, homebridge will not start?

homebridge-hue won't return the list of discovered accessories to homebridge, until all whitelisted / lightlisted resrources are available, effectively delaying the full startup of the HAP server.  Homebridge will be running on the Raspberry, but it won't yet be visible to HomeKit.  You'll see messages in the log:
```
Jan 02 23:47:57 pi2 homebridge[646]: [2018-1-2 23:47:57] Loading 1 platforms...
Jan 02 23:47:57 pi2 homebridge[646]: [2018-1-2 23:47:57] [Hue] Initializing Hue platform...
Jan 02 23:47:57 pi2 homebridge[646]: [2018-1-2 23:47:57] [Hue] homebridge-hue v0.5.57, node v8.9.3, homebridge v0.4.33
Jan 02 23:48:01 pi2 homebridge[646]: [2018-1-2 23:48:01] [Hue] pi2: deCONZ gateway v2.4.99, api v1.0.5
Jan 02 23:48:01 pi2 homebridge[646]: [2018-1-2 23:48:01] [Hue] pi2: /resourcelinks/1: /lights/101: not available
Jan 02 23:48:01 pi2 homebridge[646]: [2018-1-2 23:48:01] [Hue] pi2: gateway not yet initialised - wait 1 minute
Jan 02 23:49:02 pi2 homebridge[646]: [2018-1-2 23:49:02] [Hue] pi2: /resourcelinks/1: /lights/221: not available
Jan 02 23:49:02 pi2 homebridge[646]: [2018-1-2 23:49:02] [Hue] pi2: gateway not yet initialised - wait 1 minute
Jan 02 23:50:02 pi2 homebridge[646]: [2018-1-2 23:50:02] [Hue] pi2: 91 accessories
Jan 02 23:50:02 pi2 homebridge[646]: [2018-1-2 23:50:02] [Hue] Initializing platform accessory 'pi2'...
Jan 02 23:50:02 pi2 homebridge[646]: [2018-1-2 23:50:02] [Hue] pi2: 2 services
Jan 02 23:50:02 pi2 homebridge[646]: [2018-1-2 23:50:02] [Hue] Initializing platform accessory ...
...
Jan 02 23:50:03 pi2 homebridge[646]: Scan this code with your HomeKit app on your iOS device to pair with Homebridge:
...
Jan 02 23:50:03 pi2 homebridge[646]: Or enter this code with your HomeKit app on your iOS device to pair with Homebridge:
...
Jan 02 23:50:03 pi2 homebridge[646]: [2018-1-2 23:50:03] Homebridge is running on port 51826.
Jan 02 23:50:03 pi2 homebridge[646]: [2018-1-2 23:50:03] [Hue] masked debug info dumped to /home/pi/.homebridge/homebridge-hue.json.gz
```
When specifying your gateway(s) (and bridge(s)) in config.json, homebridge-hue already delays the start of homebridge until all gateways/bridges have been found.  So now it should be safe to start homebridge automatically on boot, irrespective of the timing of starting deCONZ.
